init/haulagedb.py

```diff
@@ -0,0 +1,127 @@
+                     passwd=db_pass,
+		     	 	 db=dbname)
+cursor = db.cursor()
+
+command = sys.argv[1]
+
+#########################################################################
+############### OPTION ONE: ADD A USER TO THE DATABASE ##################
+#########################################################################
+if (command == "add"):
+	imsi = sys.argv[2]
+	msisdn = sys.argv[3]
+	ip = sys.argv[4]
+
+	# TODO: error-handling? Check if imsi/msisdn/ip already in system?
+	print("haulagedb: adding user " + str(imsi))
+
+	commit_str = "INSERT INTO customers (imsi, msisdn) VALUES ('" + imsi + "', '" + msisdn + "')"
+	cursor.execute(commit_str)
+
+	# ip = GENERATE IP ADDRESS?!?
+	commit_str = "INSERT INTO static_ips (imsi, ip) VALUES ('" + imsi + "', '" + ip + "')"
+	cursor.execute(commit_str)
+
+#########################################################################
+############### OPTION TWO: REMOVE USER FROM THE DATABASE ###############
+#########################################################################
+elif (command == "remove"):
+	imsi = sys.argv[2]
+
+	print("haulagedb: removing user " + str(imsi))
+
+	commit_str = "DELETE FROM customers WHERE imsi = " + imsi
+	cursor.execute(commit_str)
+
+	commit_str = "DELETE FROM static_ips WHERE imsi = " + imsi
+	cursor.execute(commit_str)
+
+#########################################################################
+############### OPTION THREE: TOPUP (ADD BALANCE TO USER) ###############
+#########################################################################
+elif (command == "topup"):
+	imsi = sys.argv[2]
+	amount = decimal.Decimal(sys.argv[3])
+	old_balance = 0
+	new_balance = 0
+
+	#STEP ONE: query information
+	commit_str = "SELECT data_balance FROM customers WHERE imsi = " + imsi + " FOR UPDATE"
+	numrows = cursor.execute(commit_str)
+	if (numrows == 0):
+		print("haulagedb error: imsi " + str(imsi) + " does not exist!")
+		exit()
+
+	for row in cursor:
+		old_balance = decimal.Decimal(row[0])
+		new_balance = amount + old_balance
+
+	# STEP TWO: prompt for confirmation
+	promptstr = "haulagedb: topup user " + str(imsi) + " add " + str(amount) + " bytes to current balance of " + str(old_balance) + " bytes to create new data_balance of " + str(new_balance) + "? [Y/n] "
+	while True:
+		answer = input(promptstr)
+		if (answer == 'y' or answer == 'Y' or answer == ''):
+			print("haulagedb: updating user " + str(imsi) + " setting new data_balance to " + str(new_balance))
+			commit_str = "UPDATE customers SET data_balance = " + str(new_balance) + " WHERE imsi = " + imsi
+			cursor.execute(commit_str)
+			break
+		if (answer == 'n' or answer == 'N'):
+			print("haulagedb: cancelling topup operation\n")
+			break
+
+#########################################################################
+############### OPTION FOUR: DISABLE A USER (AND ZERO-OUT BALANCE???) ###
+#########################################################################
+# elif (command == "disable"):
```

Comment:
I don't like the dead code here... if we don't want these functions let's just get rid of them?